snuba/query/query.py

```diff
@@ -62,6 +96,58 @@ def __init__(self, body: MutableMapping[str, Any], data_source: RelationalSource
         self.__data_source = data_source
         self.__prewhere_conditions: Sequence[Condition] = []
 
+        # New data model
+        # TODO: Provide a better typing for this.
+        self.__selected_columns = selected_columns or []
+        self.__aggregations = aggregations or []
+        self.__array_join = array_join
+        self.__condition = condition
+        self.__groupby = groupby or []
+        self.__having = having
+        self.__order_by = order_by or []
+
+    def get_all_expressions(self) -> Iterable[Expression]:
+        """
+        Returns an expression container that iterates over all the expressions
+        in the query no matter which level of nesting they are at.
+        The ExpressionContainer can be used to traverse the expressions in the
+        tree.
+        """
+        return chain(*[
+            chain(*self.__selected_columns),
+            chain(*self.__aggregations),
+            self.__array_join if self.__array_join else [],
+            self.__condition if self.__condition else [],
+            chain(*self.__groupby),
+            self.__having if self.__having else [],
+            chain(*map(lambda orderby: orderby.node, self.__order_by)),
+        ])
+
+    def transform_expression(
+        self,
+        func: Callable[[Expression], Expression],
+    ) -> None:
+        """
+        Transforms in place the current query object by applying a transformation
+        function to all expressions contained in this query
+        """
+
+        def transform_expression_list(expressions: Sequence[Expression]):
+            return list(
+                map(lambda exp: exp.transform(func), expressions),
+            )
+
+        self.__selected_columns = transform_expression_list(self.__selected_columns)
+        self.__aggregations = transform_expression_list(self.__aggregations)
```

Comment:
Does this error in mypy? I'd assume it should, since this is assigning `Sequence[Expression]` to `Sequence[Aggregation]`.

Reply:
You are right, I need to make this function generic. I did not run a full mypy --strict after this last change.